src/main/java/com/internship/backend/service/ReservationService.java

```diff
@@ -55,20 +55,30 @@ public Reservation addReservation(Reservation reservation) throws ReservationAlr
         if(reservation.getEndTime().getHour() - reservation.getStartTime().getHour() != 2){
             throw new InvalidDateException("The reservation duration must be exactly 2 hours");
         }
+    }
 
-        if (isValidReservation((reservation)))
-        {
-            return reservationRepository.save(reservation);
+    private void dateValidation(NewDate startDate, NewDate endDate) throws InvalidDateException {
+        if (startDate.getMonth() != endDate.getMonth()) {
+            throw new InvalidDateException("The start date and end date must be within the same month");
         }
-        else{
-            throw new ReservationAlreadyExists("Reservation intersects another one");
+
+        if(startDate.getDay() != endDate.getDay()){
+            throw new InvalidDateException("The start date and end date must be within the same day");
+        }
+
+        if(startDate.getYear() != endDate.getYear()){
+            throw new InvalidDateException("The start date and end date must be within the same year");
+        }
+
+        if(endDate.getHour() - startDate.getHour() != 2){
+            throw new InvalidDateException("The reservation duration must be exactly 2 hours");
         }
     }
 
-    public boolean isValidReservation(Reservation reservation){
+    private boolean isValidReservation(Reservation reservation){
```

Comment:
i think this method can be rewritten in a more functional way : 
return reservationRepository.findAll().stream()
                .filter(elem -> elem.getTennisCourtId() == reservation.getTennisCourtId())
                .noneMatch(elem -> reservation.getStartTime().isBefore(elem.getEndTime()) && reservation.getEndTime().isAfter(elem.getStartTime()));